### ISSUE TITLE
[feat] New Component: Icon list

### ISSUE THREAD

**Issue by SirenaBorracha** [trussworks/react-uswds#1234]
# [Icon list](https://designsystem.digital.gov/components/icon-list)

## USWDS Version

[2.11.0](https://github.com/uswds/uswds/releases/tag/v2.11.0)

## Description

An icon list reinforces the meaning and visibility of individual list items with a leading icon.

## Requirements (proposed)

**Prop Interface**

```JSX
interface IconListProps = {
  className?: string
  children: React.ReactElement<IconListItemProps>[]
} & JSX.IntrinsicElements['ul']
```

```JSX
interface IconListItemProps = {
  className?: string
  children: React.ReactNode
} & JSX.IntrinsicElements['li']
```

```JSX
interface IconListContentProps = {
  className?: string
  children: React.ReactNode
} & JSX.IntrinsicElements['div']
```

```JSX
interface IconListHeaderProps = {
  type: string // if we want to use the pattern est in ProcessListHeading
  className?: string
  children?: React.ReactNode
} &  JSX.IntrinsicElements[TBD]
```

## Example Implementation (proposed)

```JSX
return (
  <IconList>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">Donate cash when possible.</IconListHeader>
        <p>
          Financial contributions to recognized disaster relief organizations...
        </p>
      </IconListContent>
    </IconListItem>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">
          Confirm what donations are needed.
        </IconListHeader>
        <p>
          Unneeded and unsolicited goods burden local organizations’ ability...
        </p>
      </IconListContent>
    </IconListItem>
    <IconListItem>
      <Icon name="icon-name" />
      <IconListContent>
        <IconListHeader type="h3">
          Talk to trusted organizations about volunteering.
        </IconListHeader>
        <p>
          Financial contributions to recognized disaster relief organizations...
        </p>
      </IconListContent>
    </IconListItem>
  </IconList>
)
```

**More Details**

- `<ul>`root element
- Three variants to take into consideration
-   usa-icon-list--[color] - any one of the theme colors listed on the color utilities page
-   usa-icon-list--size-[size] - theme font size: 2xs, xs, sm, md, lg, xl, or 2xl
-   [responsive_variant]:usa-icon-list--size-[size] - Add a responsive breakpoint prefix separated with a : to target a utility at a responsive breakpoint and higher, following a mobile-first methodology.

## Other Considerations

- The responsive variant may be something to put in another story as an enhancement or it might be super easy, I leave that research to the implementor

**Comment by SirenaBorracha** [comment]
Blocked by #992
